### ISSUE TITLE
error compiling pandoc-mode.el: arithmetic range error

### ISSUE THREAD

**Issue by dunn** [joostkremers/pandoc-mode#50]
May be related to #41.

Running `emacs --batch -Q --directory /private/tmp/pandoc-mode20150801-4013-i0k1o4 --directory /usr/local/opt/dash/share/emacs/site-lisp/dash --directory /usr/local/opt/hydra/share/emacs/site-lisp/hydra -f batch-byte-compile pandoc-mode.el` gives the following compile error:

```
pandoc-mode.el:1055:1:Error: Arithmetic range error: "ceiling", 1.0e+INF
```

Full logs, mostly warnings about Hydra: https://gist.github.com/dunn/b7e2095d25a80119e295

**Comment by joostkremers** [comment]
This seems to be related to the fact that `pandoc--tabulate` (in `pandoc-mode-utils.el`) uses the value returned by `frame-width` to compute the column widths of the hydras. At least, when I replace the relevant call to `frame-width` with a number, compilation is unproblematic.

Since in batch-mode there is no frame, it's probably not surprising that `frame-width` doesn't return a sensible value. Strangely, though, if I do

```
emacs --batch -Q --eval '(message "%s" (frame-width))'
```

I get `10

**Comment by dunn** [comment]
Oh, it was because I was running it in a narrow tmux frame?

**Comment by joostkremers** [comment]
No, it was because the compile command was running in batch mode. (I executed your command on a very wide tmux frame and got the same result.)

I thought I had posted a comment explaining the issue, but it's not appearing here, so I probably forgot to click the comment button... So let me explain again: `pandoc--tabulate` (the function in which the error occurs) uses the function `frame-width` to calculate how many rows and columns are required for a hydra menu (in this particular case for the extensions). In batch mode, `frame-width` appears to return a standard value of 10 (at least according to my experiments), which is too narrow for even one column. Because of the way the columns and rows are calculated, the error you reported results.

The fix simply checks for this situation and provides a (more or less) sane default. More or less, because the hydras wouldn't work anyway on a frame that's too narrow, but the situation shouldn't occur when `pandoc-mode` is used.
